lib/checker.js

```diff
@@ -1457,7 +1449,7 @@ function handleCallExpression(scope,callExprNode,isSelfRecursivePTC = false) {
 						let argSignature = typeSignatures.get(argType);
 
 						if (paramTypeID != "unknown") {
-							if (!typesMatch(argType,paramType)) {
+							if (!typesMatch(argType,paramType) && paramTypeID != "any") {
```

Comment:
You think this and line 372 are safe to change to use `isAssignmentAllowed`? They don't actually deal with assignment so maybe a more generic function name would be appropriate. Maybe to change `isAssignmentAllowed` to `typesCompatible`, that way it would function similar to `typesMatch` but more permissive.